cfme/tests/infrastructure/test_host.py

```diff
@@ -421,10 +421,21 @@ def test_infrastructure_hosts_navigation_after_download(
         1738664
 
     """
-    hosts_view = navigate_to(provider.collections.hosts, "All")
+    if hosts_collection == "provider":
+        hosts_view = navigate_to(provider.collections.hosts, "All")
+    elif hosts_collection == "appliance":
+        hosts_view = navigate_to(appliance.collections.hosts, "All")
     hosts_view.toolbar.download.item_select(report_format)
     if report_format == "Print or export as PDF":
         handle_extra_tabs(hosts_view)
     hosts_view.navigation.select("Compute")
-    provider_view = provider.create_view(InfraProviderDetailsView)
-    assert provider_view.is_displayed
+    if hosts_collection == "provider":
+        provider_view = provider.create_view(InfraProviderDetailsView)
+        assert provider_view.is_displayed
+    elif hosts_collection == "appliance":
+        assert hosts_view.is_displayed
+
+
+@pytest.fixture(params=["provider", "appliance"])
+def hosts_collection(request):
```

Comment:
Why use this method over a standard parametrization? Are you concerned about the test order? By standard parametrization I mean,

```python
@pytest.mark.parametrize("hosts_collection", ["provider", "appliance"])
```

At the top of the test function.

Reply:
Fixtures must be parameterized using this format AFAIK. The test is already paramtererised. This way we get the 3 cases in the test each run for provider and appliance, 6 tests total for each provider.

Reply:
You're right about fixtures having to be parametrized using this format. But this parametrizing doesn't need a fixture. If you put `@pytest.mark.parametrize("hosts_collection", ["provider", "appliance"])`, you should get the same 6 test cases (for each provider) that you do from from parametrized fixture. You can have two (or more) parametrizations on a single test case and pytest will create a "test matrix" for all the parameters.

Note that the ordering may be different.